pkg/virt-controller/watch/vm.go

```diff
@@ -3294,6 +3294,11 @@ func (c *VMController) handleMemoryHotplugRequest(vm *virtv1.VirtualMachine, vmi
 		return nil
 	}
 
+	if !vmi.IsMigratable() {
```

Comment:
This should be higher in the stack

Reply:
It's actually the first thing that gets checked after a change in memory is detected.

Reply:
I mean this is common for cpu/memory/affinity... Shouldn't it be before calling `handleMemoryHotplugRequest`

Reply:
Yes but we cannot do it right now, I was planning to work on that as part of the interface work for live update handlers. The issue right now is that you cannot tell whether a hotplug has been requested unless you call the handlers, so this check has to be done in the handlers for the time being.

Otherwise you'd set RestartRequired regardless of whether the VM owner is trying to change anything.

tldr; we have lots to do in this area.

Reply:
Ah, you are right... I was dreaming...

Reply:
Just looking at the future :+1: